File: modules/common-model/src/main/java/com/shterneregen/securelan/common/model/FileTransferRequest.java
package com.shterneregen.securelan.common.model;

import java.util.Objects;

public record FileTransferRequest(
        String transferId,
        String senderId,
        String recipientId,
        String fileName,
        long fileSize
) {
    public FileTransferRequest {
        Objects.requireNonNull(transferId, "transferId must not be null");
        Objects.requireNonNull(senderId, "senderId must not be null");
        Objects.requireNonNull(recipientId, "recipientId must not be null");
        Objects.requireNonNull(fileName, "fileName must not be null");
        if (fileName.isBlank()) {
            throw new IllegalArgumentException("fileName must not be blank");
        }
        if (fileSize < 0) {
            throw new IllegalArgumentException("fileSize must not be negative");
        }
    }
}

File: modules/common-model/src/main/kotlin/com/shterneregen/securelan/common/model/FileTransferRequest.kt
package com.shterneregen.securelan.common.model

import java.util.Objects

@JvmRecord
data class FileTransferRequest(
    val transferId: String,
    val senderId: String,
    val recipientId: String,
    val fileName: String,
    val fileSize: Long,
) {
    init {
        Objects.requireNonNull(transferId, "transferId must not be null")
        Objects.requireNonNull(senderId, "senderId must not be null")
        Objects.requireNonNull(recipientId, "recipientId must not be null")
        Objects.requireNonNull(fileName, "fileName must not be null")
        require(fileName.isNotBlank()) { "fileName must not be blank" }
        require(fileSize >= 0) { "fileSize must not be negative" }
    }
}
